microsoft-edge/devtools-guide-chromium/whats-new/2023/04/devtools-112.md

```diff
@@ -0,0 +1,176 @@
+
+<!-- Subtitle: Non-simple selectors can be common culprits for long-running Recalculate Styles events in the Performance tool. Take a quick snapshot in the CSS Overview tool to see your non-simple selectors. -->
+
+The **CSS Overview** tool has a new section, called **Non-simple selectors** which displays a list of non-simple CSS selectors when you take an overview of your webpage's CSS.  This list of non-simple selectors in the **Performance** tool provides a quick way to identify common culprits for long-running **Recalculate Styles** events:
+
+![The new, Non-simple selectors list in the CSS Overview tool](./devtools-112-images/non-simple-selectors-in-css-overview.png)
+
+You can try to change these selectors so that they are more specific and match fewer elements, reducing the amount of time the browser takes to calculate styles in the following cases:
+
+- When elements are added to or removed from the DOM.
+- When an element's attributes are changed, such as `class` or `id`.
+- When user input occurs, such as a mouse move or a change of focus, which can affect `:hover` rules.
+
+You can analyze the impact of your changes and individual selector performance with the **Selector Stats** feature in the **Performance** tool.  Before beginning profiling:
+
+1. In DevTools, go to the **CSS Overview** tool.
+
+1. If the **Capture overview** button is shown, click it.
+
+1. Scroll down to the **Non-simple selectors** section, and then click the **Performance** button:
+
+   ![The Performance button in the 'Non-simple selectors' section](./devtools-112-images/performance-button.png)
+
+   The **Performance** tool appears.
+
+1. Click the **Capture settings** (![Capture settings icon](./devtools-112-images/capture-settings-icon.png)) icon.
+
+1. Select the **Enable advanced rendering instrumentation (slow)** checkbox.
+
+   ![The 'Enable advanced rendering instrumentation' checkbox](./devtools-112-images/enable-advanced-rendering.png)
+
+See also:
+* [Optimize CSS styles with the CSS Overview tool](../../../css/css-overview-tool.md)
+* [Analyze selector performance during Recalculate Style events](../../../evaluate-performance/selector-stats.md)
+
+
+<!-- ====================================================================== -->
+## The JSON Viewer in Microsoft Edge supports code folding
+
+<!-- Subtitle: Use code folding when looking at JSON to collapse and expand objects and arrays. -->
+
+Starting with Microsoft Edge 110, you have access to an experimental feature for viewing JSON directly in the browser: the **JSON Viewer**.  Previously, you may have used browser extensions or text editors for viewing JSON, but now you can inspect JSON payloads from the network or from disk directly in Microsoft Edge.  Starting in Microsoft Edge 112, the JSON Viewer now supports code folding.
+
+<!-- ![The code folding setting in Microsoft Edge's JSON Viewer](./devtools-112-images/json-viewer-code-folding-setting.png) -->
+<!-- delete png? -->
```

Comment:
@zoherghadyali fyi i deleted this png, the other is sufficient.  I think it was this What's New entry where the terse UI tip was too terse and abstract to follow (wasn't able to visualize UI), and steps were in reverse usage order (click foo control in bar section in Baz tool) so I rewrote to standard verbose steps.  

One of the entries may have ended up too verbose, I need to review rendered result.